src/HyperFormula.ts

```diff
@@ -4288,13 +4288,10 @@ export class HyperFormula implements TypedEmitter {
    * @category Instance
    */
   public destroy(): void {
-    this.dependencyGraph.destroy()
-    this.columnSearch.destroy()
-    this.evaluator.destroy()
-    this._parser.destroy()
-    this._lazilyTransformingAstService.destroy()
-    this._stats.destroy()
-    this._crudOperations.clearClipboard()
+    for(const key of Object.keys(this)) {
+      delete (this as Record<string, any>)[key]
+    }
+    Object.setPrototypeOf(this, null)
```

Comment:
This will end up with an error being thrown `xxx is not a function` if someone will try to call methods on a destroyed instance. It is a good solution but maybe HOT has a better example. As @budnix pointed we replace all the functions with `postMortem` method: https://github.com/handsontable/handsontable/blob/b0a09bde9a9d4164bee4f28cd047fa843f42a7a9/src/core.js#L4006-L4016

And the `postMortem` method displays a warning:
https://github.com/handsontable/handsontable/blob/b0a09bde9a9d4164bee4f28cd047fa843f42a7a9/src/core.js#L4043-L4047

Reply:
good idea